packages/sado-connect/src/lib/signPsbt.ts

```diff
@@ -0,0 +1,69 @@
+import { Psbt } from "bitcoinjs-lib";
+import { ordit } from "@sadoprotocol/ordit-sdk";
+import { Network, Wallet } from "../providers/SadoContext";
+
+interface SignPsbtOptionsParams {
+  finalize?: boolean;
+  extractTx?: boolean;
+}
+
+interface SignPsbtParams {
+  wallet: Wallet;
+  network: Network;
+  psbt: Psbt;
+  options?: SignPsbtOptionsParams;
+}
+
+interface SignPsbtReturn {
+  rawTxHex: string | null;
+  psbt: {
+    hex: string;
+    base64: string;
+  };
+}
+
+/**
+ * @description accept wallet type and calls the right ordit function to sign the psbt.
+ * @param wallet
+ * @param network
+ * @param psbt
+ * @param options
+ */
+export default async function signPsbt({
+  wallet,
+  network,
+  psbt,
+  options,
+}: SignPsbtParams): Promise<SignPsbtReturn> {
+  const { finalize, extractTx } = options;
```

Comment:
`finalize` is optional in `SignPsbtOptionsParams`, but it is destructured here. It will throw an error if `finalize` is not passed in.

Same goes for `extractTx`